src/user-has-access-react.component.tsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { getCurrentUser } from "@openmrs/esm-api";
+
+export default function UserHasAccessReact(props) {
```

Comment:
I think this function should go into esm-api, since that's where auth related logic is going to be (as you and i discussed over slack)

Reply:
Do you think the React component should go into esm-api? Or just the hasAccess component as you recommend below? I was under the impression that esm-api should not be importing frameworks.

Reply:
@joeldenning  @jonathandick why do we need a react component to this feature? Isn't the `userHasAccess` method sufficient to display or hide a feature depending on the privileges?

Reply:
oh got it, it's a higher order component, @jonathandick I think this component should be defined in the esm-api probably in a folder called hoc

Reply:
The problem (I think) with putting into esm-api is that we then need to "reactify" the repo, meaning add in all the support for typescript, jsx/tsx, etc. It seems cleaner to leave esm-api as a pure JS module. That's why I put the react component itself in a module already using react.

Reply:
> The problem (I think) with putting into esm-api is that we then need to "reactify" the repo, meaning add in all the support for typescript, jsx/tsx, etc. It seems cleaner to leave esm-api as a pure JS module. That's why I put the react component itself in a module already using react.

I'd rather put all auth logic together. The esm-api repo already uses typescript. In order to use react, just do the following:
1. `npm install --save-dev @babel/preset-react`
2. Modify babelrc to [have babel preset react in it](https://github.com/openmrs/openmrs-esm-login/blob/master/.babelrc#L4).

Since react is in the import map, it's available to any modules that want to use it.

Reply:
Will we also put in angular module here as well? Is that possible to have both in this repo?

Reply:
Also, i had to install react :
`npm install --save-dev @babel/preset-react`